charts/dbaas-operator/templates/mongodbprovider.yaml

```diff
@@ -14,7 +14,7 @@ spec:
   user: {{ required (printf "user is required for %s" $providerName) $providerConfig.user | quote }}
   type: {{ $providerConfig.type | default "generic" | quote }}
   auth:
-    mechanism: {{ required (printf "auth.mechanism is required for %s" $providerName) $providerConfig.auth.mechanism | quote }}
-    source: {{ required (printf "auth.source is required for %s" $providerName) $providerConfig.auth.source | quote }}
-    tls: {{ required (printf "auth.tls (bool) is required for %s" $providerName) $providerConfig.auth.tls }}
+    mechanism: {{ required (printf "auth.mechanism is required for %s" $providerName) $providerConfig.auth.mechanism | default "SCRAM-SHA-1" | quote }}
```

Comment:
if we have a default then the value isn't required so we can just have something like this:
```
mechanism: {{ $providerConfig.auth.mechanism | default "SCRAM-SHA1" | quote }}
```
otherwise you can't omit the value and fall back to the default.